### PR TITLE
Dont revert genotypes after fixing

### PR DESCRIPTION
**Pull request by bbimber**
@yfarjoun Because your earlier PR isnt committed I'm opening the merge to your branch, but if you want to merge yours and then make a separate PR to master that's fine.

This does two things: 1) tryToAddVariant() shouldnt stomp over the fixed genotypes using those from the source VC, 2) adds genotypes to the test VCF, which gives coverage over this scenario.  prior to this change, that updated VCF alone would trigger this error.

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/vcf/LiftoverVcf.java [hunk 1]
```diff
@@ -331,7 +331,6 @@ private void tryToAddVariant(final VariantContext vc, final ReferenceSequence re
 
         final VariantContextBuilder builder = new VariantContextBuilder(vc);
 
-        builder.genotypes(source.getGenotypes());
```
Comment (yfarjoun) [inline src/main/java/picard/vcf/LiftoverVcf.java]: when will the genotypes get added then?

**Comment by bbimber** [comment]
LiftoverVcf line 410 or 464.  

When the new VariantContextBuilder is created on line 332, it will copy the Genotypes from vc, which is the VCB we made with the lifted coordinates.  By calling genotypes() with source.getGenotypes(), we actually revert all the upstream work to fixGenotypes() and do the opposite of what we want.  Tests are passing...see the augmented VCF in this PR and run tests with or without that line.

**Comment by bbimber** [comment]
In addition: when running the modified (or original) LiftoverVcf on a VCF with a lot of samples (~350) I regularly get GCOutOfMemoryErrors.  I strongly suspect this is due to the code constantly creating and re-creating entire VariantContext and Genotypes objects with each step.  I have a local version of LiftoverVcf that rewrites this code to do two key things:

1) While lifting the VC, pass around VariantContextBuilder and only actually create the final VariantContext once per site, instead of 2-3 times per site like now.

2) We shouldnt need to update Genotypes with every modification.  I believe this gets really expensive when you have a lot of samples.  Rather than track allele changes as a map of Allele->Allele, we only really need the list of original genotypes and list of replaced ones.  If we have this information, we can get away with a single step where we replace old/new genotypes based on index.  That saves a lot of work.

If you're willing to look at the more substantial change, this is a big performance boost and the only way I can get this tool to work on a VCF with lots of samples.

**Comment by yfarjoun** [comment]
I'm wiling to look at performance boost and bug-fixes...but not PRs that do
both at the same time. :-)

**Comment by bbimber** [comment]
Well good b/c that's not what's in this PR - maybe we can get this bugfix closed out?

**Comment by yfarjoun** [comment]
@bbimber I merged this into my branch. Looking forward to your performance improvement PR!

**Pull request merged**